app/user/admin.py

```diff
@@ -0,0 +1,100 @@
+from flask import render_template, redirect, url_for, flash, abort
+from flask_login import login_required, current_user
+from flask_wtf import FlaskForm
+from wtforms import StringField, SubmitField, PasswordField
+from wtforms.validators import DataRequired, Email, EqualTo, Optional
+from wtforms.fields.html5 import EmailField
+from app.views import ConfirmationForm, ConfirmationFormData
+from . import groups_required, user_blueprint
+from .models import User
+
+class UserEditForm(FlaskForm):
+    username = StringField('Username')
+    givenName = StringField('Given Name', validators=[DataRequired('Please enter a given name')])
+    surname = StringField('Surname', validators=[DataRequired('Please enter a surname')])
+    mail = EmailField('E-Mail', validators=[
+            DataRequired('Please enter a E-Mail address'),
+            Email('Please enter a valid E-Mail address')])
+    password = PasswordField('Password', validators=[
+            Optional(False),
+            EqualTo('confirm', 'Passwords must match')])
+    confirm = PasswordField('Repeat password')
+    submit = SubmitField('Submit')
+
+@user_blueprint.route('/admin/users')
+@login_required
+@groups_required('admin')
+def list_users():
+    users = User.query()
+    return render_template('admin/listUsers.html',
+        users = users
+    )
+
+@user_blueprint.route('/admin/user/<string:username>/delete', methods=['GET', 'POST'])
+@login_required
+@groups_required('admin')
+def delete_user(username):
+    user = User.get(username)
+    if not user:
+        flash('Invalid user name!', 'error')
+        return redirect(url_for('user.list_users'))
+
+    form = ConfirmationForm()
+    data = ConfirmationFormData('Delete user "{}"'.format(user.username),
+             'delete user "{}"'.format(user.username),
+             url_for('user.list_users'))
+    if form.validate_on_submit():
+        user.delete()
+        flash('User has been deleted', 'success')
+        return redirect(url_for('user.list_users'))
+    return render_template('confirmationPage.html',
+        form = form,
+        data = data
+    )
+
+@user_blueprint.route('/admin/user/<string:username>/edit', methods=['GET', 'POST'])
+@login_required
+def edit_user(username, directCall = True):
+    user = User.get(username)
+    if not user:
+        flash('Invalid user name!', 'error')
+        if directCall:
```

Comment:
actually it might make sense to have a [secure redirect back](http://flask.pocoo.org/snippets/62/) thing going on here.